tests/NewParseTest.cpp

```diff
@@ -97,3 +97,109 @@ TEST_F(TApp, BuiltinComplexFail) {
 
     EXPECT_THROW(run(), CLI::ArgumentMismatch);
 }
+
+// an example of custom converter that can be used to add new parsing options
+// On MSVC and possibly some other new compilers this can be a free standing function without the template
+// specialization but this is compiler dependent
+namespace CLI {
+namespace detail {
+
+template <>
+bool lexical_cast<std::pair<std::string, std::string>>(std::string input, std::pair<std::string, std::string> &output) {
+
+    auto sep = input.find_first_of(':');
+    if((sep == std::string::npos) && (sep > 0)) {
+        return false;
+    }
+    output = {input.substr(0, sep), input.substr(sep + 1)};
+    return true;
+}
+} // namespace detail
+} // namespace CLI
+
+TEST_F(TApp, custom_string_converter) {
+    std::pair<std::string, std::string> val;
+    app.add_option("-d,--dual_string", val);
+
+    args = {"-d", "string1:string2"};
+
+    run();
+    EXPECT_EQ(val.first, "string1");
+    EXPECT_EQ(val.second, "string2");
+}
+
+TEST_F(TApp, custom_string_converterFail) {
+    std::pair<std::string, std::string> val;
+    app.add_option("-d,--dual_string", val);
+
+    args = {"-d", "string2"};
+
+    EXPECT_THROW(run(), CLI::ConversionError);
+}
+
+#if defined(__has_include)
+#if __has_include(<regex>)
```

Comment:
Does this work if you are using libstdc++ from GCC 4.8? libstdc++ 4.8 does have `<regex>`, but it's non-functional. Also, if you mix stdlibs, this will fail as well, even on a newer g++ or clang. CLI11 is forgiving if you mismatch stdlibs and compilers. See <https://stackoverflow.com/a/41186162/2402816>. It might be better to optionally enable this test from CMake, etc.

Reply:
For GCC the check worked since gcc 4.8 doesn't have __has_include.  I think the clang that used an older standard library was failing due to that issue.  I changed the check on the complex test to use the CLI11_CPP14 check.  That isn't exactly accurate since it does work on C++11 but the check for the other conditions is probably too complex to be worthwhile

i tried on ubuntu 14.04 with Clang 3.5 and gcc 4.8.  I ran into a few issues with cmake 3.5 though in the build process.    Did you want those as a separate PR or just include in this one?

Reply:
Separate PR would be nice, incase this takes a while to get right. But same PR is also acceptable.